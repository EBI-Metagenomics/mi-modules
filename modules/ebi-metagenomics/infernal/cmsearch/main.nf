
process INFERNAL_CMSEARCH {

    tag "$meta.id"

    label 'process_low'

    conda "bioconda::infernal=1.1.4"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/infernal:1.1.4--pl5321hec16e2b_1':
        'biocontainers/infernal:1.1.4--pl5321hec16e2b_1' }"

    input:
    tuple val(meta), path(seqdb)
    path covariance_model_database

    output:
    tuple val(meta), path("*.cmsearch_matches.tbl"), emit: cmsearch_tbl
    path "versions.yml"                            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    cmsearch \\
        --cpu $task.cpus \\
        --noali \\
        --hmmonly \\
        $args \\
        -Z 1000 \\
        -o /dev/null \\
        --tblout ${seqdb.baseName}.cmsearch_matches.tbl \\
        $covariance_model_database \\
        $seqdb

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        cmsearch: \$(cmsearch -h | grep -o '^# INFERNAL [0-9.]*' | sed 's/^# INFERNAL *//')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${seqdb.baseName}.cmsearch_matches.tbl

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        cmsearch: \$(cmsearch -h | grep -o '^# INFERNAL [0-9.]*' | sed 's/^# INFERNAL *//')
    END_VERSIONS
    """
}
