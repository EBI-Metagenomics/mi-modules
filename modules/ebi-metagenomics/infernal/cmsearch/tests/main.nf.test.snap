{
    "versions": {
        "content": [
            [
                "versions.yml:md5,dbb2c5ffb7adf59524eb1f91b4beb5e2"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-02-06T13:27:08.164579859"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.cmsearch_matches.tbl.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,62d14c19b521b0beab9c874b8c9a4617"
                ],
                "cmsearch_tbl": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.cmsearch_matches.tbl.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,62d14c19b521b0beab9c874b8c9a4617"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-02-06T12:19:45.013255884"
    }
}